server/src/main/java/org/elasticsearch/search/fetch/subphase/InnerHitsPhase.java

```diff
@@ -40,7 +40,7 @@ public InnerHitsPhase(FetchPhase fetchPhase) {
 
     @Override
     public FetchSubPhaseProcessor getProcessor(FetchContext searchContext) {
-        if (searchContext.innerHits() == null || searchContext.innerHits().getInnerHits().isEmpty()) {
+        if (searchContext.skipInnerHits() || searchContext.innerHits() == null || searchContext.innerHits().getInnerHits().isEmpty()) {
```

Comment:
It might be more straightforward to skip inner hits in the query parsing, here:https://github.com/elastic/elasticsearch/blob/09146792256211b47549e162af5177309b4c3107/server/src/main/java/org/elasticsearch/search/SearchService.java#L1288 and here https://github.com/elastic/elasticsearch/blob/09146792256211b47549e162af5177309b4c3107/server/src/main/java/org/elasticsearch/search/SearchService.java#L1292

This way we don't need to propagate the info in the search context.

Reply:
That's a nice suggestion, thanks @jimczi ! Will update accordingly.  Only minor comment that I have is that we would now have multiple places of control instead of just one, on the inner hits phase itself.

Reply:
Updated in https://github.com/elastic/elasticsearch/pull/116408/commits/167605f89afd26b3deefdcb8647707fcf0c20e38